src/parameter/parameterVisitor.ts

```diff
@@ -1,6 +1,45 @@
 import { NodePath } from '@babel/traverse';
 import { types as t } from '@babel/core';
 
+/**
+ * Helper function to create a field/class decorator from a parameter decorator.
+ * Field/class decorators get three arguments: the class, the name of the method
+ * (or 'undefined' in the case of the constructor) and the position index of the
+ * parameter in the argument list.
+ * Some of this information, the index, is only available at transform time, and
+ * has to be stored. The other arguments are part of the decorator signature and
+ * will be passed to the decorator anyway. But the decorator has to be called
+ * with all three arguments at runtime, so this creates a function wrapper, which
+ * takes the target and the key, and adds the index to it.
+ *
+ * Inject() becomes function(target, key) { return Inject()(target, key, 0) }
+ *
+ * @param paramIdx the index of the parameter inside the function call
+ * @param decExpr the decorator expression, the return object of SomeParameterDecorator()
+ * @param isConstructor indicates if the key should be set to 'undefined'
+ */
+function makeParamDecorator(
```

Comment:
- `createParamDecorator` to keep it coherent
- `paramIndex`
- `decoratorExpression`
Just avoid unnecessary abbreviations